custom_components/candy/binary_sensor.py

```diff
@@ -0,0 +1,325 @@
+    HoodStatus,
+    MachineState,
+    OvenState,
+    TumbleDryerStatus,
+    DryerProgramState,
+    OvenStatus,
+    DishwasherStatus,
+    DishwasherState,
+)
+from .const import *
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TIME_MINUTES, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+from homeassistant.helpers.entity import DeviceInfo
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities
+):
+    """Set up the Candy sensors from config entry."""
+
+    config_id = config_entry.entry_id
+    coordinator = hass.data[DOMAIN][config_id][DATA_KEY_COORDINATOR]
+
+    if type(coordinator.data) is WashingMachineStatus:
+        async_add_entities(
+            [
+                CandyWashingMachineBinarySensor(coordinator, config_id)
+            ]
+        )
+    elif type(coordinator.data) is TumbleDryerStatus:
+        async_add_entities(
+            [
+                CandyTumbleDryerBinarySensor(coordinator, config_id)
+            ]
+        )
+    elif type(coordinator.data) is OvenStatus:
+        async_add_entities(
+            [
+                CandyOvenBinarySensor(coordinator, config_id)
+            ]
+        )
+    elif type(coordinator.data) is HoodStatus:
+        async_add_entities(
+            [
+                CandyHoodBinarySensor(coordinator, config_id)
+            ]
+        )
+    elif type(coordinator.data) is FridgeStatus:
+        async_add_entities(
+            [
+                CandyFridgeBinarySensor(coordinator, config_id)
+            ]
+        )
+    elif type(coordinator.data) is HobStatus:
+        async_add_entities(
+            [
+                CandyHobBinarySensor(coordinator, config_id)
+            ]
+        )                
+    elif type(coordinator.data) is DishwasherStatus:
+        async_add_entities(
+            [
+                CandyDishwasherBinarySensor(coordinator, config_id)
+            ]
+        )
+    else:
+        raise Exception(f"Unable to determine machine type: {coordinator.data}")
+
+
+class CandyBaseBinarySensor(CoordinatorEntity, BinarySensorEntity):
+    def __init__(self, coordinator: DataUpdateCoordinator, config_id: str):
+        super().__init__(coordinator)
+        self.config_id = config_id
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.config_id)},
+            name=self.device_name(),
+            manufacturer="Candy",
+            suggested_area=self.suggested_area(),
+        )
+
+    @abstractmethod
+    def device_name(self) -> str:
+        pass
+
+    @abstractmethod
+    def suggested_area(self) -> str:
+        pass
+
+
+class CandyWashingMachineBinarySensor(CandyBaseBinarySensor):
```

Comment:
I don't see the value in creating for a binary sensor for washing machines. In reality, when the appliance is not on, it's phisically turned off and it becomes `Unavailable` in Home Assistant.
Creating a binary sensor for other appliances makes sense though!

Reply:
This sensors just tell us if apliance work or not.
This information is useful in some UI integrations.

Example: I use  [minimalistic-area-card](https://github.com/junalmeida/homeassistant-minimalistic-area-card).  This card just show me white icon when appliance is on and gray icon when is of. But  homeassistant-minimalistic-area-card can use only binary sensor for this feature.

Reply:
Sorry, but I feel that this should be solved outside of this integration, such as with a custom template binary sensor.